core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaMerchantOffersTranslator.java

```diff
@@ -144,12 +152,22 @@ public static void openMerchant(GeyserSession session, ClientboundMerchantOffers
         session.sendUpstreamPacket(updateTradePacket);
     }
 
-    private static NbtMap getItemTag(GeyserSession session, ItemStack stack, int specialPrice) {
+    private static NbtMap getItemTag(GeyserSession session, ItemStack stack, int specialPrice, int demand, float priceMultiplier) {
+        ItemMapping mapping = session.getItemMappings().getMapping(stack);
```

Comment:
Looks like this would be called twice since it's also called in the below getItemTag? Can we avoid that?